File: App/app/src/main/java/com/example/MobileGPT/MobileGPTSpeechRecognizer.java
package com.example.MobileGPT;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;
import android.os.Looper;
import android.speech.RecognitionListener;
import android.speech.RecognizerIntent;
import android.speech.SpeechRecognizer;
import android.speech.tts.TextToSpeech;
import android.speech.tts.UtteranceProgressListener;
import android.util.Log;

import java.util.ArrayList;
import java.util.Locale;
// 在文件开头的导入区域添加
import java.util.HashMap;
public class MobileGPTSpeechRecognizer implements TextToSpeech.OnInitListener {
    private static final String TAG = "MobileGPT_SPEECH";
    private Context mContext;
    private TextToSpeech mTts;
    private UtteranceProgressListener ttsListener;
    public boolean sttOn = false;
    public MobileGPTSpeechRecognizer(Context context) {
        mContext = context;
        sttOn = false;
        mTts = new TextToSpeech(mContext, this);
        ttsListener = new UtteranceProgressListener() {
            @Override
            public void onStart(String s) {
            }
            @Override
            public void onDone(String s) {
            }

            @Override
            public void onError(String s) {
            }
        };
        mTts.setOnUtteranceProgressListener(ttsListener);
    }

    @Override
    public void onInit(int status) {
        if (status == TextToSpeech.SUCCESS) {
            // Set your preferred language and other TTS settings here
            // Set language to English (US)
            mTts.setLanguage(Locale.US);
//            mTts.setLanguage(Locale.getDefault());
        } else {
            // Handle TTS initialization failure
        }
    }

    public void speak(String text, boolean needResponse) {

//        HashMap<String, String> params = new HashMap<>();
//        mTts.speak(text, TextToSpeech.QUEUE_FLUSH, params, "tts_id");
//        if (needResponse)
//            sttOn = true;



        mTts.speak(text, TextToSpeech.QUEUE_FLUSH, null, "tts_id");
        if (needResponse)
            sttOn = true;
    }
}

File: App/app/src/main/java/Agent/MobileGPTSpeechRecognizer.kt
package Agent

import android.content.Context
import android.speech.tts.TextToSpeech
import android.speech.tts.UtteranceProgressListener
import java.util.Locale

/**
 * MobileGPT语音识别器类
 */
class MobileGPTSpeechRecognizer(mContext: Context) : TextToSpeech.OnInitListener {
    
    companion object;

    private var mTts: TextToSpeech
    private var ttsListener: UtteranceProgressListener
    
    /**
     * 语音转文本是否开启
     */
    var sttOn = false
    
    init {
        sttOn = false
        mTts = TextToSpeech(mContext, this)
        ttsListener = object : UtteranceProgressListener() {
            override fun onStart(utteranceId: String?) {
                // TTS开始播放时的回调
            }
            
            override fun onDone(utteranceId: String?) {
                // TTS播放完成时的回调
            }
            
            override fun onError(utteranceId: String?) {
                // TTS播放出错时的回调
            }
        }
        mTts.setOnUtteranceProgressListener(ttsListener)
    }
    
    /**
     * TTS初始化回调
     * @param status 初始化状态
     */
    override fun onInit(status: Int) {
        if (status == TextToSpeech.SUCCESS) {
            // 在这里设置您的首选语言和其他TTS设置
            // 设置语言为英语（美国）
            mTts.language = Locale.US
            // mTts.language = Locale.getDefault()
        } else {
            // 处理TTS初始化失败
        }
    }
    
    /**
     * 播放语音
     * @param text 要播放的文本
     * @param needResponse 是否需要响应
     */
    fun speak(text: String, needResponse: Boolean) {
        mTts.speak(text, TextToSpeech.QUEUE_FLUSH, null, "tts_id")
        if (needResponse) {
            sttOn = true
        }
    }
}